client/src/components/Login/index.jsx

```diff
@@ -6,8 +6,31 @@ export default class extends Component {
   state = {
     email: '',
     password: '',
+    msg:''
   };
+  handelChange = ({ target: { name, value } }) => {
+    if(name ==="email"&& value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i))
```

Comment:
@Jamalat-shamallakh It might be helpful in the future if you take this regex value and put it into its own file, and then import it and use it here:
```
import validEmailRegex from '../shared/validEmailRegex'
...
...
if(name ==="email"&& value.match(validEmailRegex))
```
Or you could make an isEmailValid function in your file and just call that.
This would help you for the next place that you need to validate emails in this app (it's a common thing to do).

Reply:
it's a good idea.